lib/Driver/ToolChains/Clang.cpp

```diff
@@ -4351,6 +4349,12 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
   if (Args.hasFlag(options::OPT_fvectorize, VectorizeAliasOption,
                    options::OPT_fno_vectorize, EnableVec))
     CmdArgs.push_back("-vectorize-loops");
+  
+  if (Args.getLastArg(options::OPT_fveclib))
+    Args.AddLastArg(CmdArgs, options::OPT_fveclib);
+  else if (Args.hasFlag(options::OPT_fvectorize, VectorizeAliasOption,
+                        options::OPT_fno_vectorize, EnableVec))
+    CmdArgs.push_back("-fveclib=PGMATH");
```

Comment:
What happens if someone is using a version of LLVM without support for -fveclib=PGMATH?

Reply:
LLVM generates an error as it would with any option which it does not recognize.

Reply:
While flang has its own fork of llvm, flang still works with upstream llvm as well.  The changes for DWARF, for example, are enabled with a compile-time #ifdef in flang-compiler/flang.  Likely we need a similar mechanism here until we know that all of the upstream guides to building flang (not just blogs, but spack etc) are updated to refer to flang-compiler/flang.

Reply:
There is already a CMAKE variable that flang builders can use to include Flang-specific LLVM extensions. (FLANG_LLVM_EXTENSIONS)

Reply:
I don't agree that -fvectorize should imply extra hidden arguments. The user should be in control of the kind of vectorization and the library calls to be used and these should remain independent.

Reply:
I thought we wanted `-fveclib=PGMATH` to be the default. I can undo this.

Reply:
We do want PGMATH to be the default but the flang ecosystem is not ready for that b/c we still want to build with llvm from llvm.org or pre-installed llvm in a distro; we need to move these changes upstream.  Unlike SVML, libpgmath is always available with flang so making it the default will not have any negative user impact.

Reply:
Please add FLANG_LLVM_EXTENSIONS to flang-driver in a subsequent commit; soon please.